### PR TITLE
Fix int8 requant

### PR DESCRIPTION
**Pull request by tpoisonooo**
* fix int8 requant in `convolution_arm.cpp: 1163`,
* add int8 requant test

`top_blob_g` defined but not used, compile warning should be treated as error.

### CODE REVIEW

File: src/layer/arm/convolution_arm.cpp [hunk 1]
```diff
@@ -1137,6 +1137,7 @@ int Convolution_arm::forward_int8_arm(const Mat& bottom_blob, Mat& top_blob, con
         }
         else
         {
+
```
Comment (nihui) [inline src/layer/arm/convolution_arm.cpp]: useless whitespace changes

File: src/layer/arm/convolution_arm.cpp [hunk 2]
```diff
@@ -1149,8 +1150,7 @@ int Convolution_arm::forward_int8_arm(const Mat& bottom_blob, Mat& top_blob, con
             opt_g.blob_allocator = top_blob.allocator;
 
             Mat top_blob_tm_g = top_blob_tm.channel_range(p, 1);
-            Mat top_blob_g = top_blob.channel_range(p, 1);
-
+            Mat top_blob_g    = top_blob.channel_range(p, 1);
```
Comment (nihui) [inline src/layer/arm/convolution_arm.cpp]: useless whitespace changes

File: src/layer/arm/gemm_symm_int8.h [hunk 3]
```diff
@@ -38,7 +38,7 @@
     int n1 = n;\
     n = ntmp;
 
-#define PRINT_MATRIX 0
+#define PRINT_MATRIX 0
```
Comment (nihui) [inline src/layer/arm/gemm_symm_int8.h]: useless whitespace changes

File: src/layer/arm/gemm_symm_int8.h [hunk 4]
```diff
@@ -2678,8 +2706,12 @@ void int8kernel(void* dst, const int8_t* sa, const int8_t* sb, int m, int k, int
             default:
                 break;
         }
+
```
Comment (nihui) [inline src/layer/arm/gemm_symm_int8.h]: useless whitespace changes

File: tests/testutil.h [hunk 5]
```diff
@@ -138,6 +152,23 @@ static int CompareMat(const ncnn::Mat& a, const ncnn::Mat& b, float epsilon = 0.
     return 0;
 }
 
+static int CompareMat(const ncnn::Mat& a, const ncnn::Mat& b, float epsilon = 0.001)
+{
+    CHECK_MEMBER(elemsize)
+
+    if (4 == a.elemsize || 16 == a.elemsize)
```
Comment (nihui) [inline tests/testutil.h]: if (a.elemsize / a.elempack == 4)

File: tests/testutil.h [hunk 6]
```diff
@@ -138,6 +152,23 @@ static int CompareMat(const ncnn::Mat& a, const ncnn::Mat& b, float epsilon = 0.
     return 0;
 }
 
+static int CompareMat(const ncnn::Mat& a, const ncnn::Mat& b, float epsilon = 0.001)
+{
+    CHECK_MEMBER(elemsize)
```
Comment (nihui) [inline tests/testutil.h]: elemsize member checking is done in Compare function, this one is duplicated

**Comment by nihui** [comment]
```
std::function<void(ncnn::Layer*)> prehook = nullptr
```
please use function pointer, we don't force c++11 features at the moment
```
typedef void (*prehook_func)(ncnn::Layer*);
```

**Comment by tpoisonooo** [comment]
o

File: src/layer/arm/convolution_arm.cpp [hunk 7]
```diff
@@ -1149,7 +1149,7 @@ int Convolution_arm::forward_int8_arm(const Mat& bottom_blob, Mat& top_blob, con
             opt_g.blob_allocator = top_blob.allocator;
 
             Mat top_blob_tm_g = top_blob_tm.channel_range(p, 1);
-            Mat top_blob_g    = top_blob.channel_range(p, 1);
```
Comment (tpoisonooo) [inline src/layer/arm/convolution_arm.cpp]: = 号对齐它不好看么。

Comment (nihui) [reply src/layer/arm/convolution_arm.cpp]: 单独看，好看
一起看，大家都不对齐，你对齐，你就不好看了

Comment (kalcohol) [reply src/layer/arm/convolution_arm.cpp]: 大家帅，就我不

**Pull request merged**